### PR TITLE
FLPROD-796: Create snippets api client

### PR DESCRIPTION
**Pull request by WowVeryLogin**
Adding client for snippets API to use in v4 terraform with tests

### CODE REVIEW

File: snippets.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	tp, err := mw.CreateFormField("metadata")
+	if err != nil {
+		return "", nil, err
+	}
+	_, err = tp.Write([]byte(fmt.Sprintf(`{"main_module":%s}`, request.MainFile)))
```
Comment (jacobbednarz) [inline snippets.go]: don't want to use a struct here and have stdlib handle the marshaling?

**Review by jacobbednarz — COMMENTED** [review]
seems fine to me with a non-blocking comment on the marshaling.

**Comment by jacobbednarz** [comment]
(ignore the changelog check, it's fine)

File: snippets.go (hunk 1 above)
Comment (WowVeryLogin) [reply snippets.go]: Fixed

File: .changelog/3458.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
+snippets: add CRUD support
+```
+
+```release-note:enhancement
+snippet_rules: add CRUD support
```
Comment (jacobbednarz) [inline .changelog/3458.txt]:
```suggestion
snippets_rules: add CRUD support
```

**Pull request merged**